Make CUDA/NCCL plugins link against the existing ucc libraries.

Bart Oldeman, 2022-08-02
diff -ur ucc-1.0.0.orig/src/components/mc/cuda/Makefile.am ucc-1.0.0/src/components/mc/cuda/Makefile.am
--- ucc-1.0.0.orig/src/components/mc/cuda/Makefile.am	2022-04-15 12:43:33.000000000 +0000
+++ ucc-1.0.0/src/components/mc/cuda/Makefile.am	2022-08-02 12:13:59.332795976 +0000
@@ -15,7 +15,7 @@
 libucc_mc_cuda_la_CFLAGS   = $(BASE_CFLAGS)
 libucc_mc_cuda_la_LDFLAGS  = -version-info $(SOVERSION) --as-needed $(CUDA_LDFLAGS)
 libucc_mc_cuda_la_LIBADD   = $(CUDA_LIBS)                      \
-                             $(UCC_TOP_BUILDDIR)/src/libucc.la \
+                             -lucc \
                              kernel/libucc_mc_cuda_kernels.la
 
 include $(top_srcdir)/config/module.am
diff -ur ucc-1.0.0.orig/src/components/tl/nccl/Makefile.am ucc-1.0.0/src/components/tl/nccl/Makefile.am
--- ucc-1.0.0.orig/src/components/tl/nccl/Makefile.am	2022-04-15 12:43:33.000000000 +0000
+++ ucc-1.0.0/src/components/tl/nccl/Makefile.am	2022-08-02 12:13:59.334795989 +0000
@@ -21,6 +21,6 @@
 libucc_tl_nccl_la_CPPFLAGS = $(AM_CPPFLAGS) $(BASE_CPPFLAGS) $(CUDA_CPPFLAGS) $(NCCL_CPPFLAGS)
 libucc_tl_nccl_la_CFLAGS   = $(BASE_CFLAGS)
 libucc_tl_nccl_la_LDFLAGS  = -version-info $(SOVERSION) --as-needed $(CUDA_LDFLAGS) $(NCCL_LDFLAGS)
-libucc_tl_nccl_la_LIBADD   = $(CUDA_LIBS) $(NCCL_LIBADD) $(UCC_TOP_BUILDDIR)/src/libucc.la
+libucc_tl_nccl_la_LIBADD   = $(CUDA_LIBS) $(NCCL_LIBADD) -lucc
 
 include $(top_srcdir)/config/module.am
